src/autopas/selectors/tuningStrategy/PredictiveTuning.h

```diff
@@ -610,14 +656,31 @@ void PredictiveTuning::selectOptimalConfiguration() {
   _currentConfig = _searchSpace.find(optimum->first);
 
   // sanity check
-  if (_currentConfig == _searchSpace.end() or _currentConfig == _optimalSearchSpace.end()) {
+  if (_currentConfig == _searchSpace.end() or _currentConfig == _optimalSearchSpace.end() or
+      _currentConfig == _tooLongNotTestedSearchSpace.end()) {
     autopas::utils::ExceptionHandler::exception(
         "PredicitveTuning: Optimal configuration not found in list of configurations!");
   }
 
+  if (_relativeRangeForBlacklist != 0 and not _notTestedYet.empty()) selectBlackList();
+
   AutoPasLog(debug, "Selected Configuration {}", _currentConfig->toString());
 }
 
+void PredictiveTuning::selectBlackList() {
+  auto optimumTime = _traversalTimesStorage[getCurrentConfiguration()].back().second;
+  std::set<Configuration> oldNotTestedYet = _notTestedYet;
+
+  for (const auto &configuration : oldNotTestedYet) {
+    if (_lastTest[configuration] == _tuningIterationsCounter) {
+      if (_traversalTimesStorage[configuration].back().second / optimumTime > _relativeRangeForBlacklist) {
+        _blacklistSearchSpace.emplace(configuration);
+      }
+      _notTestedYet.erase(configuration);
```

Comment:
if you use a real iterator (`auto iter = set.begin()`) you can call `iter = set.erase(iter)` and directly work on `_notTestedYet` without copying it.